servers/rendering/renderer_rd/shaders/effects/copy.glsl

```diff
@@ -46,26 +46,38 @@ layout(set = 0, binding = 0) uniform samplerCubeArray source_color;
 #elif defined(MODE_CUBEMAP_TO_PANORAMA)
 layout(set = 0, binding = 0) uniform samplerCube source_color;
 #elif !defined(MODE_SET_COLOR)
-layout(set = 0, binding = 0) uniform sampler2D source_color;
+layout(set = 0, binding = 0) uniform sampler2DArray source_color;
 #endif
 
 #ifdef GLOW_USE_AUTO_EXPOSURE
 layout(set = 1, binding = 0) uniform sampler2D source_auto_exposure;
 #endif
 
 #if defined(MODE_LINEARIZE_DEPTH_COPY) || defined(MODE_SIMPLE_COPY_DEPTH)
-layout(r32f, set = 3, binding = 0) uniform restrict writeonly image2D dest_buffer;
+layout(r32f, set = 3, binding = 0) uniform restrict writeonly image2DArray dest_buffer;
 #elif defined(DST_IMAGE_8BIT)
-layout(rgba8, set = 3, binding = 0) uniform restrict writeonly image2D dest_buffer;
+layout(rgba8, set = 3, binding = 0) uniform restrict writeonly image2DArray dest_buffer;
 #else
-layout(rgba32f, set = 3, binding = 0) uniform restrict writeonly image2D dest_buffer;
+layout(rgba32f, set = 3, binding = 0) uniform restrict writeonly image2DArray dest_buffer;
 #endif
 
 #ifdef MODE_GAUSSIAN_BLUR
 shared vec4 local_cache[256];
 shared vec4 temp_cache[128];
 #endif
 
+vec4 texelFetch(sampler2DArray tex, ivec2 pos, int lod) {
```

Comment:
I know this works, but its probably quite confusing to read, I would use a special name for this like fetch_texel() or something like this

Reply:
Done: texelFetch -> fetch_texel; imageStore -> store_image; textureLod -> sample_texture_lod

the only places I did not replace to wrapper functions were sampling the `source_color` for cubemap (when it is of type `samplerCube` and `samplerCubeArray`), and sampling the `source_auto_exposure` (`sampler2D`)

I have tested some of the effects, but I probably missed some cases, since this shader does a lot.